api/src/main/java/io/strimzi/api/kafka/model/Probe.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.strimzi.crdgenerator.annotations.Minimum;
+import io.sundr.builder.annotations.Buildable;
+import io.vertx.core.cli.annotations.DefaultValue;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A representation of the configurable aspect of a probe (used for health checks).
+ */
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = true,
+        builderPackage = "io.strimzi.api.kafka.model"
+)
+public class Probe {
+    private int initialDelaySeconds = 15;
+    private int timeoutSeconds = 5;
+    private Map<String, Object> additionalProperties = new HashMap<>(0);
+
+    @Description("The initial delay before first the health is first checked.")
+    @Minimum(0)
+    @DefaultValue("15")
+    public int getInitialDelaySeconds() {
+        return initialDelaySeconds;
+    }
+
+    public void setInitialDelaySeconds(int initialDelaySeconds) {
+        this.initialDelaySeconds = initialDelaySeconds;
+    }
+
+    @Description("The timeout for each attempted health check.")
+    @Minimum(0)
+    @DefaultValue("5")
+    public int getTimeoutSeconds() {
+        return timeoutSeconds;
+    }
+
+    public void setTimeoutSeconds(int timeoutSeconds) {
+        this.timeoutSeconds = timeoutSeconds;
+    }
+
+    @JsonAnyGetter
+    public Map<String, Object> getAdditionalProperties() {
+        return this.additionalProperties;
+    }
+
+    @JsonAnySetter
+    public void setAdditionalProperty(String name, Object value) {
+        this.additionalProperties.put(name, value);
+    }
+}
```

Comment:
The fabric8 `Probe` is the full thing, with properties for `httpGet`, `exec`, `tpcSocket` as well as the delay, timeout. If we reused it people could mistakenly set those things and think it would make a difference. Really the CO is in charge of what the probe actually does, and we just want to give the user some control over how it gets run. Maybe for this reason the name Probe is confusing and we should name it `Healthcheck`, or something else.

Reply:
Yes at this point. I would prefer a different name as `Healthcheck`.